### ISSUE TITLE
CASL Prisma does not work as expected when defining conditions on relation

### ISSUE THREAD

**Issue by 5t111111** [stalniy/casl#735]
## Describe the bug

Let’s say you have two basic Prisma model `User` and `Post`, there users have their own posts. `Post` models have two publishing status called `true` and `false`.

If you want to get all users who have "published" posts, you can do that via Prisma query like the following:

```ts
const users = await prisma.user.findMany({
  where: {
    posts: {
      some: { 
        published: true
      }
    },
  },
});
```

If you make users can read other users with published posts, I expect the following rule work via CASL rule condition:

```ts
can("read", "User", { posts: { some: { published: true } } });
```

However it does not work, users cannot read other users even when they have "published" posts.

The CASL Prisma document says:

> when defining conditions on relation, always specify one of operators (every, none, some, is or isNot)

That rule has `some` operator in it, so it should satisfy the rule for "conditions on relation".

I think this behavior is unexpected issue because I found the following comments by exploring the existing issues:

https://github.com/stalniy/casl/issues/161#issuecomment-1222201378
https://github.com/stalniy/casl/issues/161#issuecomment-1222641607

## To Reproduce

Minimal reproducible example repository

https://github.com/5t111111/casl-prisma-repro

It includes:

- It has two basic model `User` and `Post`
- Defining conditions on relation described in the previous section
- Script for checking if rules work as expected
- Debugging via `relevantRuleFor`

## Expected behavior

The rules work with conditions on relation if you use the same condition as `where` clause query condition.

## Workaround

The only workaround I have found so far is finding all users by the same filter as the rule, after that define rule using that result without using relation condition:

```ts
 const users = await prisma.user.findMany({
   where: {
     posts: { some: { published: true } },
   },
 });
 can("read", "User", { id: { in: users.map((u) => u.id) } });
```

## CASL Version

- @casl/ability: 6.3.3
- @casl/prisma: 1.3.3

## Environment

- Node.js: 18.14.1
- TypeScript: 4.9.5
- Prisma: 4.10.1

**Comment by stalniy** [comment]
Hi 

Thank you for super easy to follow issue and README file in corresponding repository! The issue is that CASL by itself doesn't fetch anything for you. as bob has no relation to posts (in RAM), CASL sees there are no posts on that record and returns false. 

But if you change your code to this:

<img width="512" alt="Screenshot 2023-02-25 at 00 01 33" src="https://user-images.githubusercontent.com/1159966/221301608-5643a1d2-24a2-40d5-8f60-9a5de5b1a722.png">

Then you will get `true`.

**Comment by 5t111111** [comment]
Thank you!

I was able to confirm that the suggested modifications work as expected, and also understand how CASL works under the hood.

It is indeed the first time using CASL in our products, it is really amazing! I have found using CASL will solve most of our authorization complexity with its clear and testable design. It is also very familiar to me because I used to use cancancan for a while. Thank you again for providing great authorization library.
